### ISSUE TITLE
Default filter watermark error

### ISSUE THREAD

**Issue by mrzasan** [theatlantic/thumbor-video-engine#2]
When I enable `thumbor-video-engine`  and add `thumbor_video_engine.filters.format`, `thumbor_video_engine.filters.still` to Filters Setting, thumbor's default filter `wartermark` could not  load the background image（`yll.jpg`）, but load  watermark pic (`water_logo.png`) twice .
Here is my url:`http://minio_host/unsafe/filters:watermark(images/water_logo.png,-10,-10,50)/thumbor-docker/2020/01/17/yll.jpg`
And my setting:
![image](https://user-images.githubusercontent.com/2859944/77158598-9ac35d00-6ade-11ea-9a79-9f1536f7c6f2.png)
![image](https://user-images.githubusercontent.com/2859944/77158704-cf371900-6ade-11ea-8603-ae95bfde2905.png)
And the setting result:
![image](https://user-images.githubusercontent.com/2859944/77158762-ea098d80-6ade-11ea-8131-974fba97111e.png)
Disable the `thumbor-video-engine` and reset filters ,thumbor's default filters work well.
**ENVS**
```
Package               Version
--------------------- -----------
backports-abc         0.5
boto                  2.42.0
botocore              1.14.5
colour                0.1.5
contextlib2           0.6.0.post1
dateutils             0.6.6
derpconf              0.8.3
docutils              0.15.2
envtpl                0.4.1
futures               3.3.0
graphicsmagick-engine 0.1.1
Jinja2                2.10.3
jmespath              0.9.4
libthumbor            1.3.2
MarkupSafe            1.1.1
numpy                 1.16.6
opencv-engine         1.0.1
pgmagick              0.7.5
piexif                1.1.3
Pillow                5.1.0
pip                   19.3.1
pycryptodome          3.9.4
pycurl                7.43.0.4
pylibmc               1.6.1
pymongo               3.10.1
pyres                 1.5
python-dateutil       2.8.1
pytz                  2019.3
raven                 5.15.0
redis                 3.3.11
remotecv              2.2.2
setproctitle          1.1.10
setuptools            42.0.2
shortuuid             0.5.0
simplejson            3.17.0
singledispatch        3.4.0.3
six                   1.14.0
statsd                3.3.0
tc-aws                6.2.12
tc-core               0.4.1
tc-mongodb            5.1.0
tc-redis              1.0.1
tc-shortener          0.2.2
thumbor               6.7.0
thumbor-memcached     5.1.0
thumbor-video-engine  1.0.2
tornado               5.1.1
tornado-botocore      1.5.0
urllib3               1.25.7
virtualenv            16.7.9
webcolors             1.10
wheel                 0.33.6
```

I upgrade thumbor to V6.7.5 also failed.🤣

**Comment by fdintino** [comment]
Could you enable debug logging and post the output you get when you request an image?

**Comment by mrzasan** [comment]
> Could you enable debug logging and post the output you get when you request an image?

`
2020-03-21 17:50:03 thumbor:DEBUG METRICS: inc: response.status.400:1,
      handler_name    : Core Media Video,
2020-03-21 17:50:02 thumbor:DEBUG METRICS: inc: storage.hit:1,
2020-03-21 17:50:02 thumbor:DEBUG METRICS: inc: response.count:1,
2020-03-21 17:50:02 thumbor:DEBUG Setting still frame at position 0,
2020-03-21 17:50:02 thumbor:DEBUG Setting engine to thumbor_video_engine.engines.ffmpeg (extension .mp4),
2020-03-21 17:50:02 thumbor:DEBUG Running `/usr/bin/ffmpeg -hide_banner -i /tmp/tmpvHij3u.mp4 -ss 0 -frames:v 1 -y /tmp/tmp_A2enA.png`,
2020-03-21 17:50:03 thumbor:DEBUG Input #0, mov,mp4,m4a,3gp,3g2,mj2, from '/tmp/tmpvHij3u.mp4':,
  Metadata:,
    major_brand     : mp42,
    minor_version   : 1,
    creation_time   : 2020-03-20T13:39:17.000000Z,
  Duration: 00:00:10.04, start: 0.000000, bitrate: 781 kb/s,
    Stream #0:0(und): Video: h264 (High) (avc1 / 0x31637661), yuv420p(tv, bt709), 720x1280 [SAR 1:1 DAR 9:16], 643 kb/s, 24.80 fps, 25 tbr, 12800 tbn, 50 tbc (default),
    Metadata:,
      creation_time   : 2020-03-20T13:39:17.000000Z,
    Stream #0:1(und): Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, stereo, fltp, 127 kb/s (default),
      handler_name    : Core Media Audio,
Stream mapping:,
  Stream #0:0 -> #0:0 (h264 (native) -> png (native)),
Press [q] to stop, [?] for help,
Output #0, image2, to '/tmp/tmp_A2enA.png':,
    compatible_brands: isommp41mp42,
    encoder         : Lavf58.20.100,
    Stream #0:0(und): Video: png, rgb24, 720x1280 [SAR 1:1 DAR 9:16], q=2-31, 200 kb/s, 25 fps, 25 tbn, 25 tbc (default),
      encoder         : Lavc58.35.100 png,
frame=    1 fps=0.0 q=-0.0 Lsize=N/A time=00:00:00.04 bitrate=N/A speed=0.201x    ,
2020-03-21 17:50:03 PIL.PngImagePlugin:DEBUG STREAM 'IHDR' 16 13,
2020-03-21 17:50:03 PIL.PngImagePlugin:DEBUG STREAM 'pHYs' 41 9,
2020-03-21 17:50:03 PIL.PngImagePlugin:DEBUG STREAM 'cHRM' 62 32,
2020-03-21 17:50:03 PIL.PngImagePlugin:DEBUG STREAM 'gAMA' 106 4,
2020-03-21 17:50:03 thumbor:DEBUG Setting engine to thumbor.engines.pil (extension None),
2020-03-21 17:50:03 PIL.PngImagePlugin:DEBUG STREAM 'IDAT' 122 4096,
2020-03-21 17:50:03 PIL.PngImagePlugin:DEBUG STREAM 'iCCP' 41 3113,
2020-03-21 17:50:03 PIL.PngImagePlugin:DEBUG iCCP profile name 'ICC Profile',
2020-03-21 17:50:03 PIL.PngImagePlugin:DEBUG Compression method 0,
2020-03-21 17:50:03 PIL.PngImagePlugin:DEBUG STREAM 'pHYs' 3166 9,
2020-03-21 17:50:03 PIL.PngImagePlugin:DEBUG STREAM 'IDAT' 3187 8671,
2020-03-21 17:50:03 thumbor:DEBUG Image format specified as .jpg.,
2020-03-21 17:50:03 thumbor:DEBUG Content Type of image/jpeg detected.,
2020-03-21 17:50:03 tornado.access:INFO 304 GET /unsafe/smart/filters:still():watermark(images/water_logo.png,10,10,10)/jyt/niu.MP4 (123.144.117.93) 501.09ms,
2020-03-21 17:50:03 thumbor:DEBUG METRICS: timing: response.time:489,
2020-03-21 17:50:03 thumbor:DEBUG METRICS: timing: response.time.304:489,
2020-03-21 17:50:03 thumbor:DEBUG METRICS: inc: response.status.304:1,
2020-03-21 17:50:03 thumbor:DEBUG METRICS: inc: response.format.jpg:1,
2020-03-21 17:50:03 thumbor:DEBUG METRICS: timing: response.time.jpg:489,
2020-03-21 17:50:03 thumbor:DEBUG METRICS: inc: response.count:1,
2020-03-21 17:50:03 thumbor:DEBUG METRICS: inc: response.bytes.jpg:9779,
2020-03-21 17:50:03 thumbor:WARNING URL does not have hash or unsafe, or has both: /favicon.ico,
2020-03-21 17:50:03 tornado.access:WARNING 400 GET /favicon.ico (123.144.117.93) 4.96ms,
2020-03-21 17:50:03 thumbor:DEBUG METRICS: timing: response.time:2,
2020-03-21 17:50:03 thumbor:DEBUG METRICS: timing: response.time.400:2`

**Comment by chrischip** [comment]
I am still having exactly the same issue with thumbor-video-engine ver1.2.2 and thumbor ver7.0.6

**Comment by chrischip** [comment]
the issue is watermark module recursively use the Engine object and the image buffer get overwritten by the watermark. I am not sure what is the best way to resolve this. If you have the same issue and want a quick fix, you could replace this line in thumbor\filters\watermark.py:

`self.watermark_engine = self.context.modules.engine.__class__(self.context)`

with this:

`self.watermark_engine = self.context.modules.importer.image_engine(self.context)`

Then watermark will work again.

**Comment by fdintino** [comment]
This should be fixed in the 1.2.3 release, which is now available on PyPI.
